scantool/scantool_850.c

```diff
@@ -678,21 +678,58 @@ static enum cli_retval cmd_850_ping(int argc, UNUSED(char **argv)) {
 	return CMD_OK;
 }
 
+#define CLAMPED_LOOKUP(table,index) table[(((unsigned)(index))>=(sizeof(table)/sizeof(*table)-1))?(sizeof(table)/sizeof(table[0])-1):((unsigned)(index))]
```

Comment:
an almost free improvement is to use our already existing ARRAY_SIZE macro in there

Reply:
I've changed it to:
`#define CLAMPED_LOOKUP(table,index) table[MIN(ARRAY_SIZE(table)-1,(unsigned)(index))]`